addons/repair/models/repair.py

```diff
@@ -251,10 +265,9 @@ def action_repair_confirm(self):
         return True
 
     def action_repair_cancel(self):
-        if self.filtered(lambda repair: repair.state == 'done'):
-            raise UserError(_("Cannot cancel completed repairs."))
-        if any(repair.invoiced for repair in self):
-            raise UserError(_('The repair order is already invoiced.'))
+        invoice_to_cancel = self.filtered(lambda repair: repair.invoice_id.state == 'draft').mapped('invoice_id')
```

Comment:
```suggestion
        invoice_to_cancel = self.filtered(lambda repair: repair.invoice_state == 'draft').mapped('invoice_id')
```

We have field at https://github.com/odoo/odoo/pull/45888/files#diff-75d9849e302e0edb197ad769026c0e81R114